flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/serializer/AvroSchemaConvertor.java

```diff
@@ -41,8 +54,12 @@
  */
 public class AvroSchemaConvertor {
 
+    private AvroSchemaConvertor() {}
+
     private static final Logger LOG = LoggerFactory.getLogger(AvroSchemaConvertor.class);
 
+    private static final ConcurrentMap<Schema, DataType> map = new ConcurrentHashMap<>();
```

Comment:
Lets use an elaborate name like `conversionMemorizationMap` or `avroSchemaToDataTypeMap`

Reply:
Whats the maximum size this map can reach? Is it possible that we can get a schema large/deep enough to cause heap memory issues?

Reply:
One way to test is creating the largest possible schema while ensuring that its data element is within the maximum row size supported by BQ write API. If such a schema is successfully memorized by this map then you're good, else we need to figure it out.